Unread notification badge

The regular Discord client for Linux has an unread notification badge that shows up in my KDE task manager, if you turn on the "Enable Unread Message Badge" setting. [It looks like this](https://i.imgur.com/Zss0uxA.png) (the colour varies depending on if your DE has a colour scheme.) However, it only works if libunity is present on the system (almost certainly because Discord only tested their software on Ubuntu, but I digress.) If libunity SOs not present then no badges will ever be shown in the task manager, regardless of whether that setting is on or off.

Keeping libunity and its dependencies up-to-date on a system that doesn't ship with it is quite a hassle. Is there anything flatpak can do to implement this feature?

Oh duh - I should've done that up front. Okay, with submodules included I have a build error with `dee`'s makefile:
```
/usr/bin/mkdir: cannot create directory ‘/usr/lib/python3.9/site-packages/gi’: Read-only file system
make[3]: *** [Makefile:379: install-pygioverridesPYTHON] Error 1
make[3]: Leaving directory '/run/build/dee/bindings/python'
make[2]: *** [Makefile:472: install-am] Error 2
make[2]: Leaving directory '/run/build/dee/bindings/python'
make[1]: *** [Makefile:406: install-recursive] Error 1
make[1]: Leaving directory '/run/build/dee/bindings'
make: *** [Makefile:527: install-recursive] Error 1
Error: module dee: Child process exited with code 2
```
I don't have python 3.9, so I guess this is provided by flatpak's build system, right? Looks to me like it's trying to install a Python package by directly writing files into python 3.9's home dir but isn't being given write permission.

--- a/com.discordapp.Discord.json
+++ b/com.discordapp.Discord.json
@@ -26,7 +26,10 @@
         "--talk-name=com.canonical.Unity",
         "--own-name=org.kde.*",
         "--env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons",
-        "--env=XDG_CURRENT_SESSION=KDE"
+        /* Electron checks if it's running on Unity/KDE before using the UnityLauncherAPI. To make it work on other Desktops too, we need to pretend we are on KDE. */
+        "--env=XDG_CURRENT_SESSION=KDE",
+        /* Make moving files to trash work */
+        "--env=ELECTRON_TRASH=gio"
     ],
     "modules": [
         {

--- a/com.discordapp.Discord.json
+++ b/com.discordapp.Discord.json
@@ -26,7 +26,7 @@
         "--talk-name=com.canonical.Unity",
         "--own-name=org.kde.*",
         "--env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons",
-        /* Electron checks if it's running on Unity/KDE before using the UnityLauncherAPI. To make it work on other Desktops too, we need to pretend we are on KDE. */
+        /* Electron checks if it's running on Unity or KDE before using the UnityLauncherAPI. To make it work on other Desktops too, we need to pretend we are on KDE. */
         "--env=XDG_CURRENT_SESSION=KDE",
         /* Make moving files to trash work because it chooses which to use based on XDG_CURRENT_SESSION.  */
         "--env=ELECTRON_TRASH=gio"

--- a/com.discordapp.Discord.json
+++ b/com.discordapp.Discord.json
@@ -28,7 +28,7 @@
         "--env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons",
         /* Electron checks if it's running on Unity/KDE before using the UnityLauncherAPI. To make it work on other Desktops too, we need to pretend we are on KDE. */
         "--env=XDG_CURRENT_SESSION=KDE",
-        /* Make moving files to trash work */
+        /* Make moving files to trash work because it chooses which to use based on XDG_CURRENT_SESSION.  */
         "--env=ELECTRON_TRASH=gio"
     ],
     "modules": [

--- a/com.discordapp.Discord.json
+++ b/com.discordapp.Discord.json
@@ -23,9 +23,10 @@
         "--talk-name=org.kde.StatusNotifierWatcher",
         "--talk-name=com.canonical.AppMenu.Registrar",
         "--talk-name=com.canonical.indicator.application",
-        "--talk-name=com.canonical.Unity.LauncherEntry",
+        "--talk-name=com.canonical.Unity",
         "--own-name=org.kde.*",
-        "--env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons"
+        "--env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons",
+        "--env=XDG_CURRENT_SESSION=KDE"
     ],
     "modules": [
         {
@@ -60,7 +61,8 @@
                 "install -d /app/share/applications",
                 "sed -e 's/Icon=discord/Icon=com.discordapp.Discord/' -e 's|Exec=/usr/share/discord/Discord|Exec=com.discordapp.Discord|' /app/discord/discord.desktop > /app/share/applications/${FLATPAK_ID}.desktop",
                 "install -Dm644 /app/discord/discord.png /app/share/icons/hicolor/256x256/apps/${FLATPAK_ID}.png",
-                "install -Dm644 com.discordapp.Discord.appdata.xml /app/share/appdata/com.discordapp.Discord.appdata.xml"
+                "install -Dm644 com.discordapp.Discord.appdata.xml /app/share/appdata/com.discordapp.Discord.appdata.xml",
+                "patch-desktop-filename ${FLATPAK_DEST}/discord/resources/app.asar"
             ],
             "sources": [
                 {

